src/services/utilServices/MailClient.ts

```diff
@@ -28,16 +35,57 @@ class MailClient {
     }
 
     try {
-      await axios.post(endpoint, email, {
+      const sendMailResponse = await axios.post<MailData>(sendEndpoint, email, {
         headers: {
           Authorization: `Bearer ${this.POSTMAN_API_KEY}`,
         },
       })
+      this.verifyMail(sendMailResponse.data)
     } catch (err) {
       logger.error(`Error occurred when sending email to ${recipient}: ${err}`)
       throw new Error("Failed to send email.")
     }
   }
+
+  async verifyMail(sendMailData: MailData): Promise<void> {
+    const verfyEndpoint = `${POSTMAN_API_URL}/transactional/email/${sendMailData.id}`
+    await new Promise((res) => setTimeout(res, MAIL_VERIFICATION_DELAY))
+    const verifyMailResponse = await axios.get<MailData>(verfyEndpoint, {
+      headers: {
+        Authorization: `Bearer ${this.POSTMAN_API_KEY}`,
+      },
+    })
+    this.logEmailStatus(verifyMailResponse.data)
+  }
+
+  logEmailStatus(verifyMailData: MailData) {
+    const mailStatus = verifyMailData.status
+    switch (mailStatus) {
+      case "UNSENT":
+        logger.error(
+          `Email to ${verifyMailData.recipient} not accepted: ${verifyMailData}`
+        )
+        break
+      case "ACCEPTED":
+        logger.error(
+          `Email to ${verifyMailData.recipient} not sent: ${verifyMailData}`
+        )
+        break
+      case "SENT":
+        logger.error(
+          `Email to ${verifyMailData.recipient} not delivered: ${verifyMailData}`
+        )
+        break
+      case "BOUNCED":
+        logger.error(
+          `Email to ${verifyMailData.recipient} rejected by recipient's mail server: ${verifyMailData}`
+        )
```

Comment:
why is accepted/sent an `error`?

Reply:
`ACCEPTED` means the email is not sent. Sent does not guarantee that the email has been successfully delivered. The ideal state after 1 min should be `DELIVERED` or `OPENED` as `SENT` suggests that the email is invalid and there is no email server handling the email.

Reply:
supernit: can we don't use `logger.error` for those but rather `warn`? ie, we use `error` only for end error states and for transitions, we just `warn`. this isn't really a big deal but just a good to have so that in the event where we crawl through logs, we can sift through less

Reply:
okay, changing to warning instead